core/src/main/java/org/opensearch/sql/analysis/Analyzer.java

```diff
@@ -118,34 +121,25 @@ public LogicalPlan analyze(UnresolvedPlan unresolved, AnalysisContext context) {
 
   @Override
   public LogicalPlan visitRelation(Relation node, AnalysisContext context) {
+    QualifiedName qualifiedName = node.getTableQualifiedName();
+    CatalogSchemaIdentifierName catalogSchemaIdentifierName
+        = new CatalogSchemaIdentifierName(qualifiedName.getParts(), catalogService.getCatalogs());
+    String tableName = catalogSchemaIdentifierName.getIdentifierName();
     context.push();
     TypeEnvironment curEnv = context.peek();
-    String catalogName = getCatalogName(node);
-    String tableName = getTableName(node);
-    if (catalogName != null && !catalogService.getCatalogs().contains(catalogName)) {
-      tableName = catalogName + "." + tableName;
-      catalogName = null;
-    }
     Table table = catalogService
-        .getStorageEngine(catalogName)
-        .getTable(tableName);
+        .getStorageEngine(catalogSchemaIdentifierName.getCatalogName())
```

Comment:
Change the interface of https://github.com/opensearch-project/sql/blob/2.x/plugin/src/main/java/org/opensearch/sql/plugin/catalog/CatalogServiceImpl.java#L83 to 
`StorageEngine getStorageEngine(String catalog)` and remove the `catalog == null` check?

Reply:
ok...catalog == null check is not needed earlier as well...since if it is not in keySet..it will anyways result in defaultStorageEngine.

Reply:
will remove.